boilerplate/app/screens/DemoShowroomScreen/SectionListWithKeyboardAwareScrollView.tsx

```diff
@@ -0,0 +1,60 @@
+import { DEFAULT_BOTTOM_OFFSET } from "app/components"
+import React, { ReactElement } from "react"
+import { ScrollViewProps, SectionList, SectionListProps } from "react-native"
+import { KeyboardAwareScrollView } from "react-native-keyboard-controller"
+
+type SectionType<ItemType> = {
+  name: string
+  description: string
+  data: ItemType[]
+}
+
+type SectionListWithKeyboardAwareScrollViewProps<ItemType> = SectionListProps<ItemType> & {
+  /* Optional function to pass a custom scroll component */
+  renderScrollComponent?: (props: ScrollViewProps) => React.ReactNode
+  /* Optional bottom offset for the keyboard avoid behavior */
+  bottomOffset?: number
```

Comment:
Since this passing along a react-native-keyboard-controller prop, I think linking to the docs in the comment is useful, something like:

```
  /* Optional additional offset between TextInput bottom edge and keyboard top edge. See https://kirillzyusko.github.io/react-native-keyboard-controller/docs/api/components/keyboard-aware-scroll-view#bottomoffset */
  bottomOffset?: number

```